src/rust/graph-merger/src/main.rs

```diff
@@ -31,64 +47,151 @@ pub mod service;
 pub mod upsert_util;
 pub mod upserter;
 
```

Comment:
Is there any intent to eventually dedupe all this repeated service stuff? It's a non-negligible amount of boiler plate

Reply:
Yes, we have a story somewhere to actually build some `Service` abstractions, but that's been back-burnered until we get the Kafka migration done. I think that's the right order of operations. Some things are becoming clear--e.g. `clap` for configurations, the `kafka` library is coming together, we're getting a handle on our gRPC services in `rust-proto-new`.. but we also have some unknowns like healthchecks, circuit breakers, and service lifecycle states.

Ultimately, the idea is to eventually make some kind of a `Service` (a lot like a [Guava Service](https://github.com/google/guava/wiki/ServiceExplained)) and some kind of `Builder` we can use in e.g. `graph-merger/main.rs` or `pipeline-ingress/main.rs` to tie all the pieces together. But the technical requirements for this "thing" haven't really come into focus yet, and it would be a big waste to put a lot of effort into _actually building_ that thing until we really know what it should be. I think we're close, and I hope we can put time into that project soon.

Here's the tracking issue: https://app.zenhub.com/workspaces/grapl-6036cbd36bacff000ef314f2/issues/grapl-security/issue-tracker/909